adapter/src/dummy.rs

```diff
@@ -63,27 +63,43 @@ impl Adapter for DummyAdapter {
         Ok(is_same)
     }
 
-    fn validate_channel(&self, _channel: &Channel) -> AdapterResult<bool> {
-        // @TODO
-        Ok(true)
+    fn validate_channel(&self, channel: &Channel) -> AdapterResult<bool> {
+        match DummyAdapter::is_channel_valid(&self.config, channel) {
+            Ok(_) => Ok(true),
+            Err(e) => Err(AdapterError::InvalidChannel(format!("{}", e))),
+        }
     }
 
-    fn session_from_token(&self, _token: &str) -> AdapterResult<String> {
-        // @TODO
-        Ok("hello".to_string())
+    fn session_from_token(&mut self, token: &str) -> AdapterResult<Session> {
+        let mut identity = "";
+        for (key, val) in self.tokens_for_auth.iter() {
+            if val == token {
+                identity = key;
+            }
+        }
+
+        Ok(Session {
+            uid: identity.to_owned(),
+            era: 0,
+        })
     }
 
-    fn get_auth(&self, _validator: &ValidatorDesc) -> AdapterResult<String> {
-        // let participant = self
-        //     .participants
-        //     .iter()
-        //     .find(|&(_, participant)| participant.identity == validator);
-        // let future = match participant {
-        //     Some((_, participant)) => ok(participant.token.to_string()),
-        //     None => err(AdapterError::Authentication(
-        //         "Identity not found".to_string(),
-        //     )),
-        // };
-        Ok("auth".to_string())
+    fn get_auth(&mut self, _validator: &ValidatorDesc) -> AdapterResult<String> {
+        let who = self
+            .tokens_verified
+            .clone()
+            .into_iter()
+            .find(|(_, id)| *id == self.identity);
+
+        match who {
+            Some((id, _)) => {
+                let auth = self.tokens_for_auth.get(&id).unwrap();
```

Comment:
this `unwrap` should be `expect` so that you can explain why you make that assumption